homeassistant/components/climate/tado_v1.py

```diff
@@ -0,0 +1,341 @@
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        # pylint: disable=no-member
+        if self._min_temp:
+            return self._min_temp
+        else:
+            # get default temp from super class
+            return ClimateDevice.min_temp.fget(self)
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        #  pylint: disable=no-member
+        if self._max_temp:
+            return self._max_temp
+        else:
+            #  Get default temp from super class
+            return ClimateDevice.max_temp.fget(self)
+
+    def sensor_changed(self, entity_id, old_state, new_state):
+        #  pylint: disable=W0613
+        """Called when a depending sensor changes."""
+        if new_state is None or new_state.state is None:
+            return
+
+        self.update_state(entity_id, new_state, True)
+
+    def update_state(self, entity_type, state, update_ha):
+        """Update the internal state."""
+        if state.state == "unknown":
+            return
+
+        _LOGGER.info("%s changed to %s", entity_type, state.state)
+
+        try:
+            if entity_type.endswith("temperature"):
+                unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+
+                self._cur_temp = self.hass.config.units.temperature(
+                    float(state.state), unit)
+
+                self._target_temp = self.hass.config.units.temperature(
+                    float(state.attributes.get("setting")), unit)
+
+            elif entity_type.endswith("humidity"):
+                self._cur_humidity = float(state.state)
+
+            elif entity_type.endswith("tado mode"):
+                self._is_away = state.state == "AWAY"
+
+            elif entity_type.endswith("power"):
+                if state.state == "OFF":
+                    self._current_operation = CONST_MODE_OFF
+                    self._device_is_active = False
+                else:
+                    self._device_is_active = True
+
+            elif entity_type.endswith("overlay"):
+                #  if you set mode manualy to off, there will be an overlay
+                #  and a termination, but we want to see the mode "OFF"
+                overlay = state.state
+                termination = state.attributes.get("termination")
+
+                if overlay == "True" and self._device_is_active:
+                    #  there is an overlay the device is on
+                    self._overlay_mode = termination
+                    self._current_operation = termination
+                elif overlay == "False":
+                    #  there is no overlay, the mode will always be
+                    #  "SMART_SCHEDULE"
+                    self._overlay_mode = CONST_MODE_SMART_SCHEDULE
+                    self._current_operation = CONST_MODE_SMART_SCHEDULE
+
+            if update_ha:
+                self.schedule_update_ha_state()
+
+        except ValueError:
```

Comment:
Please do't add a try except over hole function. Only of code part that can raise the error.